tests/ignite/handlers/test_state_param_scheduler.py

```diff
@@ -0,0 +1,291 @@
+    step_state_parameter_scheduler = StepStateScheduler(
+        param_name="step_scheduled_param", initial_value=initial_value, gamma=gamma, step_size=step_size
+    )
+    step_state_parameter_scheduler.attach(engine, Events.EPOCH_COMPLETED)
+    engine.run([0] * 8, max_epochs=max_epochs)
+    torch.testing.assert_allclose(
+        getattr(engine.state, "step_scheduled_param"), initial_value * gamma ** (max_epochs // step_size)
+    )
+
+
+from bisect import bisect_right
+
+
+@pytest.mark.parametrize(
+    "max_epochs, initial_value, gamma, milestones, save_history",
+    [(3, 10, 0.99, [3, 6], True), (40, 5, 0.98, [3, 6, 9, 10, 11], True)],
+)
+def test_multistep_scheduler(
+    max_epochs, initial_value, gamma, milestones, save_history,
+):
+    engine = Engine(lambda e, b: None)
+    multi_step_state_parameter_scheduler = MultiStepStateScheduler(
+        param_name="multistep_scheduled_param", initial_value=initial_value, gamma=gamma, milestones=milestones,
+    )
+    multi_step_state_parameter_scheduler.attach(engine, Events.EPOCH_COMPLETED)
+    engine.run([0] * 8, max_epochs=max_epochs)
+    torch.testing.assert_allclose(
+        getattr(engine.state, "multistep_scheduled_param"),
+        initial_value * gamma ** bisect_right(milestones, max_epochs),
+    )
+
+
+def test_custom_scheduler():
+
+    initial_value = 10
+    gamma = 0.99
+    engine = Engine(lambda e, b: None)
+
+    def lambda_fn(event_index):
+        return initial_value * gamma ** (event_index % 9)
+
+    lambda_state_parameter_scheduler = LambdaStateScheduler(param_name="custom_scheduled_param", lambda_fn=lambda_fn,)
+    lambda_state_parameter_scheduler.attach(engine, Events.EPOCH_COMPLETED)
+    engine.run([0] * 8, max_epochs=2)
+    torch.testing.assert_allclose(getattr(engine.state, "custom_scheduled_param"), lambda_fn(2))
+    engine.run([0] * 8, max_epochs=20)
+    torch.testing.assert_allclose(getattr(engine.state, "custom_scheduled_param"), lambda_fn(20))
+
+
+@pytest.mark.parametrize(
+    "scheduler_cls,scheduler_kwargs",
+    [
+        (
+            LinearStateScheduler,
+            {
+                "param_name": "linear_scheduled_param",
+                "initial_value": 0,
+                "step_constant": 2,
+                "step_max_value": 5,
+                "max_value": 10,
+            },
+        ),
+        (ExpStateScheduler, {"param_name": "exp_scheduled_param", "initial_value": 10, "gamma": 0.99}),
+        (
+            StepStateScheduler,
+            {"param_name": "step_scheduled_param", "initial_value": 10, "gamma": 0.99, "step_size": 5},
+        ),
+        (
+            MultiStepStateScheduler,
+            {"param_name": "multistep_scheduled_param", "initial_value": 10, "gamma": 0.99, "milestones": [3, 6]},
+        ),
+        (
+            LambdaStateScheduler,
+            {"param_name": "custom_scheduled_param", "lambda_fn": lambda event_index: 10 * 0.99 ** (event_index % 9),},
+        ),
+    ],
+)
```

Comment:
Try to reduce the number of lines here as well. For example, remove trailing commas and use predefined variables as suggested above

Reply:
ok thanks @vfdev-5